packages/core/src/__tests__/ceramic-feed.test.ts

```diff
@@ -30,63 +34,92 @@ describe('Ceramic feed', () => {
+        feed1.push(s)
+      })
+      const s2 = await ceramic2.feed.aggregation.documents.pipe(take(3)).subscribe((s) => {
+        feed2.push(s)
+      })
+      const stream1 = await TileDocument.create(ceramic1, content, null, {
+        anchor: false,
+        publish: true,
+      })
+
+      // load stream on ceramic node 2
+      await TileDocument.load(ceramic2, stream1.id)
+      await stream1.update(updatedContent, null, { anchor: false, publish: true })
+      await TestUtils.delay(500)
+
+      expect(feed1.length).toEqual(2) // create + update
+      expect(feed2.length).toEqual(2) //load + pubsub update
+      expect(feed2[0].content.test).toBe(content.test)
+      expect(feed2[0].commitId).toStrictEqual(feed1[0].commitId)
+      // test pubsub propagating the update from stream1 being inside the feed
+      expect(feed2[1].commitId).toStrictEqual(feed1[1].commitId)
+      expect(feed2[1].metadata).toStrictEqual(feed1[1].metadata)
+      expect(feed2[1].content).toStrictEqual(updatedContent)
+
+      s1.unsubscribe()
+      s2.unsubscribe()
+    }, 20000)
+
+    test('add entry after anchoring stream', async () => {
+      const feed: FeedDocument[] = []
+      const s = ceramic1.feed.aggregation.documents.subscribe((s) => {
+        feed.push(s)
+      })
+
+      const stream = await TileDocument.create(
+        ceramic1,
+        { hello: `world-${Math.random()}` },
+        null,
+        {
+          anchor: true,
+        }
+      )
+      const state$ = await ceramic1.repository.load(stream.id, {})
+      // request anchor
+      await ceramic1.repository.anchor(state$, {})
+      // process anchor
+      await CoreUtils.anchorUpdate(ceramic1, stream)
+
+      expect(feed.length).toEqual(2) // 2 commits = 1 genesis commit + 1 anchor commit
+      expect(feed[0].content).toEqual(feed[1].content)
+      expect(feed[0].metadata).toEqual(feed[1].metadata)
+      expect(feed[0].commitId.equals(feed[1].commitId)).toEqual(false)
+      s.unsubscribe()
     })
+  })
 
-    // load stream on ceramic node 2
-    await TileDocument.load(ceramic2, stream1.id)
-    await stream1.update(updatedContent, null, { anchor: false, publish: true })
-    await TestUtils.delay(500)
-
-    expect(feed1.length).toEqual(2) // create + update
-    expect(feed2.length).toEqual(2) //load + pubsub update
-    expect(feed2[0].content.test).toBe(content.test)
-    expect(feed2[0].commitId).toStrictEqual(feed1[0].commitId)
-    // test pubsub propagating the update from stream1 being inside the feed
-    expect(feed2[1].commitId).toStrictEqual(feed1[1].commitId)
-    expect(feed2[1].metadata).toStrictEqual(feed1[1].metadata)
-    expect(feed2[1].content).toStrictEqual(updatedContent)
-
-    s1.unsubscribe()
-    s2.unsubscribe()
-  }, 20000)
-
-  test('add entry after creating/loading indexed model', async () => {
+  skipIfV4ShouldPass('add entry after creating/loading indexed model', async () => {
```

Comment:
so we short circuit and we never call `_indexStreamIfNeeded` when in V4 mode. I'm not sure if we should call `_updateStateIfPinned_safe` in the repository when in v4 mode, or if the process of storing/recv from recon should update this and it will work when it's implemented.

Reply:
iiiinteresting.  But even if we don't `_indexStreamIfNeeded` on the load path, I'd expect the stream to still be pinned and indexed on the write path.  If it isn't, then that's a bug.

Reply:
oh I see, but this test is creating it on a different node and then loading it on another node that has the feed.  So yeah then that makes sense since cross-node loading is disabled currently.  So yeah, this test should start passing once Recon is integrated.